app/models/support/gds/campaign.rb

```diff
@@ -4,14 +4,22 @@ module Support
   module GDS
     class Campaign
       include ActiveModel::Model
-      attr_accessor :title, :other_dept_or_agency, :signed_campaign, :start_date, :end_date, :description,
-                    :call_to_action, :success_measure, :proposed_url, :site_metadescription, :cost_of_campaign
+      attr_accessor :type_of_site, :has_read_guidance, :has_read_oasis_guidance, :signed_campaign,
+                    :start_date, :end_date, :development_start_date, :performance_review_contact_email,
+                    :government_theme, :description, :call_to_action, :proposed_url, :site_metadescription,
+                    :cost_of_campaign, :ga_contact_email
 
-      validates_presence_of :title, :signed_campaign, :start_date, :end_date, :description,
-                            :call_to_action, :success_measure, :proposed_url, :site_metadescription, :cost_of_campaign
+      validates_presence_of :type_of_site, :has_read_guidance, :has_read_oasis_guidance, :signed_campaign,
```

Comment:
Given we present a radio box with only two choices to the user could we validate that `type_of_site` only allows those two values?

Reply:
Do we need to store the `has_read_guidance` and `has_read_oasis_guidance` checkboxes?  Are they intended as gates were saying no means you can't submit the form (e.g. like checking "I have read the terms & conditions", or "I have read the EULA" we might get on signing up to a new website, or installing new software)?  We could use rails validates_confirmation_of on those attributes to not proceed if they don't tick yes.  (The doc isn't clear on this, so maybe it's ok to let it through if they haven't read those docs).

Reply:
Those two fields correspond to checkboxes in the form, where I'm using `required: true` to ask people to tick them. Once ticked, we're stripping them out in the CampaignRequestsController so they don't need to be forwarded. 

So we don't require the services of `validates_confirmation_of` since we never assign those fields. 

However, I've renamed the fields to `has_read_guidance_confirmation` and `has_read_oasis_guidance_confirmation` to make it more clear what they are.

Reply:
Arg.  I meant `validates_acceptance_of`, not `validates_confirmation_of`.  I don't *think* we can get away with `required: true` because rails checkboxes always submit a `0` or `1` so they're always filled in.  There's a hidden input that always submits the `0`, and if you check it the form also submits the `1` overwriting the `0` earlier in the form.  Of course, this assumes we're using rails checkbox helpers in the form which behave this way, maybe we're not.